source/agora/common/ManagedDatabase.d

```diff
@@ -78,4 +80,81 @@ public class ManagedDatabase
     {
         return this.database;
     }
+
+    /***************************************************************************
+
+        Begins a batch operation on all the open databases.
+
+        A beginBatch() call must be followed by either a commitBatch() or
+        a rollback() call (in case of exceptions).
+
+        The rollback() call will cancel any transactions started after the
+        beginBatch() call.
+
+        In case of crashes it is not necessary to call rollback() because
+        none of the transactions were COMMITED, and since the application
+        crashed it will have to re-start and initialize new database handles
+        which will not have any of the previously pending transactions in
+        its transaction queue.
+
+    ***************************************************************************/
+
+    public static void beginBatch () @trusted nothrow
+    {
+        foreach (db; thread_dbs)
+        {
+            try
+            {
+                db.begin();
+            }
+            catch (Exception exc)
+            {
+                // in most cases this would only trigger if there was a
```

Comment:
I will fix this up with the work that's going on in https://github.com/bpfkorea/agora/issues/836. I need all the SQLite errors to properly be handled in the binding, and then consider which cases here are fatal /  could be retried.